src/client/runner/automation/playback/select/index.js

```diff
@@ -0,0 +1,217 @@
+    }
+
+    _calculateAbsoluteEndPoint () {
+        var point = selectUtils.getSelectionPositionCoordinates(this.element, this.endPos);
+
+        if (point)
+            return point;
+
+        // NOTE: if selection ends on an invisible symbol, we should try to find the last visible selection position
+        if (domUtils.isContentEditableElement(this.element))
+            return selectUtils.getLastVisibleSelectionPosition(this.element, this.startPos, this.endPos);
+
+        return positionUtils.findCenter(this.element);
+    }
+
+    _moveToStart () {
+        selectUtils.scrollEditableElementByPoint(this.element, this.absoluteStartPoint);
+
+        this.startPoint = selectUtils.excludeElementScroll(this.element, this.absoluteStartPoint);
+
+        var moveArguments = {
+            element: document.documentElement,
+            offsetX: this.startPoint.x,
+            offsetY: this.startPoint.y
+        };
+
+        return this._move(moveArguments);
+    }
+
+    _mousedown () {
+        return cursor
+            .leftButtonDown()
+            .then(() => {
+                this.eventArgs = SelectAutomation._calculateEventArguments(this.startPoint);
+
+                // NOTE: in WebKit and IE, the mousedown event opens the select element's dropdown.
+                // Therefore, we should prevent mousedown and hide the dropdown (B236416)
+                var needCloseSelectDropDown = (browserUtils.isWebKit || browserUtils.isIE) &&
+                                              domUtils.isSelectElement(this.element);
+
+                if (needCloseSelectDropDown)
+                    this._bindMousedownHandler();
+
+                if (browserUtils.hasTouchEvents) {
+                    this.eventState.simulateDefaultBehavior = eventSimulator.touchstart(this.eventArgs.element,
+                        this.eventArgs.options);
+                }
+                else {
+                    this.eventState.simulateDefaultBehavior = eventSimulator.mousedown(this.eventArgs.element,
+                        this.eventArgs.options);
+                }
+
+
+                if (this.eventState.simulateDefaultBehavior === false) {
+                    this.eventState.simulateDefaultBehavior = needCloseSelectDropDown &&
+                                                              !this.eventState.mousedownPrevented;
+                }
+
+                return this._focus();
+            });
+    }
+
+    _focus () {
+        return new Promise(resolve => {
+            // NOTE: If the target element is a child of a contentEditable element, we need to call focus for its parent
+            var elementForFocus = domUtils.isContentEditableElement(this.element) ?
+                                  contentEditable.findContentEditableParent(this.element) : this.element;
+
+            focusBlurSandbox.focus(elementForFocus, resolve, false, true);
+        });
+    }
+
+    _moveToEnd () {
```

Comment:
Move to end of what? I guess it's the moving with the selection. We need a better name for this one